core/src/main/java/ar/com/wolox/wolmo/core/util/WolmoFileProvider.kt

```diff
@@ -38,6 +42,61 @@ import javax.inject.Inject
 @ApplicationScope
 class WolmoFileProvider @Inject constructor(private val context: Context) {
 
+    private val appName: String by lazy {
+        val applicationInfo = context.applicationInfo
+        val stringId = applicationInfo.labelRes
+        if (stringId != 0) context.getString(stringId) else applicationInfo.nonLocalizedLabel?.toString() ?: ""
+    }
+
+    private var cacheFolder: String = getTmpDirectory(context).absolutePath
+
+    private fun getTmpDirectory(context: Context) = File(context.cacheDir, appName).apply {
+        if (!exists()) {
+            mkdir()
+        }
+    }
+
+    private fun getEnvironmentFilename(folder: String, name: String, extension: String, mime: String): String? {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            val resolver = context.contentResolver
+            val contentValues = ContentValues().apply {
+                put(MediaStore.MediaColumns.DISPLAY_NAME, FILENAME_FORMAT.format(name, System.nanoTime()))
+                put(MediaStore.MediaColumns.MIME_TYPE, mime)
+                put(MediaStore.MediaColumns.RELATIVE_PATH, "$folder/$appName")
+            }
+
+            resolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, contentValues)?.let {
```

Comment:
Between `resolver` creation and this line, it's not used. we can do
`context.contentResolver.insert(....)` instead. What do you think?